### ISSUE TITLE
Update contribution guidelines - standardize param construction 

### ISSUE THREAD

**Issue by mattnolting** [patternfly/patternfly-next#499]
The param construction varies throughout components. 
Examples: 
`pf-c-alert--Modifier`
`pf-c-badge--modifier`
`btnClass`
`<div class="pf-c-content {{ modifierClass }}">`
`pf-c-dropdown--Modifier` 
`modal-box--modifiers`

The correct construction should be: `[partial]--modifiers` + `[partial]--attributes`

For example, in: `component.hbs`
```
<div class="classname{{#if component--modifier}} {{component--modifier}}{{/if}}"
  {{#if component--attribute}}
    {{{component--attribute}}}
  {{/if}}>
</div>

component--modifier relates to the partial itself, rather than the assigned class as the assigned class can change

```
Case for:
- `[component-name]--` rather than `pf-c-[component-name]--`
Using partial name (instead of classname) means that if a selector changes, you won't have to change every instance where that partial is called and modified (in example files). Also, the correct parameter name may not be immediately obvious from an example file. As a result, you have to reference the partial. A good example of this is: 
`{{#> card-body pf-c-card__content-modifier="" }}` vs `{{#> card-body card-body--modifier="" }}`

```
{{#> partial partial--modifier="[mod]" partial--attribute='id="[id]" partial--expandable="true" partial--expanded="true"}}
  content
{{/partial}}
```

rather than

```
{{#> partial pf-c-partial--modifiers="special" pf-c-partial--attributes='id="[id]" aria-label="[label]" aria-describedby="[description]"'  pf-c-partial--expandable="true" pf-c-partial--expanded="true"}}
  content
{{/partial}}
```

**Comment by srambach** [comment]
All good points and highlighting some inconsistencies as we've evolved from the time we started. 

I think I like leaving the `pf-c-` off and using just `[component-name]--` because it does remove a little clutter and differentiates handlebars parameters from style names.

We should have already been conforming to `--modifier` according to our contribution guidelines. I prefer the singular `--modifier` and `--attribute` (after all, HTML uses singular `style=""` and `class=""`, even if multiple styles or classes are defined). But I am more concerned with consistency, settling on conventions, and sticking to them.
